ddl/ddl_api.go

```diff
@@ -2779,6 +2851,37 @@ func checkColumnWithIndexConstraint(tbInfo *model.TableInfo, originalCol, newCol
 	return nil
 }
 
+func checkAutoRandom(tableInfo *model.TableInfo, originCol *table.Column, specNewColumn *ast.ColumnDef) error {
+	if !config.GetGlobalConfig().Experimental.AllowAutoRandom && containsColumnOption(specNewColumn, ast.ColumnOptionAutoRandom) {
```

Comment:
If `containsColumnOption(specNewColumn, ast.ColumnOptionAutoRandom)` is false, I think we need to return first.

Reply:
Here is a counterexample to prove early return is incorrect:
```
create table t (a int auto_random(3) primary key);
alter table t modify column a int;
```
Normally, removing the `auto_random` column option is not allowed. If we do early return, we will fail to check the above constraint.

Reply:
Got it.
How about `AllowAutoRandom` is false and the `ColumnOptionAutoRandom` of this column doesn't change? Like:
```
// AllowAutoRandom is true.
create table t (a int auto_random(3) primary key);
// AllowAutoRandom is false.
alter table t modify column a int auto_random(3) ;
```

Reply:
Since the config says it is not allowed to use `auto_random`, the second statement should report an error.